### PR TITLE
Propogating the ReassociateFpReduction flag of IREE Codegen to ConvertToLLVM.

### PR DESCRIPTION
**Pull request by vmurali**
Please use just the second commit to see just the diff with https://github.com/iree-org/iree/pull/10537

### CODE REVIEW

**Review by vmurali — COMMENTED** [review]
Addressing https://github.com/iree-org/iree/issues/10279.

Use only the second commit (to see the diff over https://github.com/iree-org/iree/pull/10537)

**Comment by vmurali** [comment]
Addressing https://github.com/iree-org/iree/issues/10279

File: compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp [hunk 1]
```diff
@@ -52,6 +52,8 @@
 namespace mlir {
 namespace iree_compiler {
 
+extern llvm::cl::opt<bool> clEnableReassociateFpReductions;
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:55]: I didnt realize we could do that. Its interesting, but maybe we can add a boolean option flag to the `ConvertToLLVM` pass and set that by passing it through `LLVMCPU/Passes.cpp`?

Comment (vmurali) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:55]: Do you mean move the definition to `ConvertToLLVM`, and the `extern` declaration to `LLVMCPU/Passes.cpp` ?

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:55]: interesting, I was not aware of this. IMO, we should add an optional flag to `ConvertToLLVM` pass, and have pass it like `createConvertToLLVMPass(clEnableReassociateFpReductions)`. Like what we've done for SplitReduction.

The Passes.cpp (i.e., pipeline constructer) has higher scope. The definition of the flag should exist in the Passes.cpp or higher layer, not in ConvertToLLVM.

Comment (vmurali) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:55]: So, two separate flags? One for `SplitReduction`, and one for `ConvertToLLVM` ? But the intent of both the passes is the same, right - reorderFpReductions.

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:55]: > we should add an optional flag to ConvertToLLVM pass, and have pass it like createConvertToLLVMPass(clEnableReassociateFpReductions).

I meant the optional flag in td file, which implies that an extra argument to `createConvertToLLVMPass`.

Comment (vmurali) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp]: > > we should add an optional flag to ConvertToLLVM pass, and have pass it like createConvertToLLVMPass(clEnableReassociateFpReductions).
> 
> I meant the optional flag in td file (i.e, the ConvertToLLVM pass).

Done

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp]: LGTM thanks!

Comment (vmurali) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp]: Can I get an LGTM for it, so that I can submit it?

**Review by hanhanW — APPROVED** [review]

**Comment by MaheshRavishankar** [comment]
Its surprising this didnt give any benefit. Is it off by default?

**Comment by vmurali** [comment]
It's off by default

**Pull request merged**